### PR TITLE
Update 2022 c2

### PR DESCRIPTION
**Pull request by JunAishima**
updates for new Bluesky conda environment including those required to make acceptance work with new env

### CODE REVIEW

File: startup/39-original_suitcase.py [hunk 1]
```diff
@@ -28,14 +26,45 @@ def conv_to_list(d):
...
+    if debug:
+        print(f"resource locations: {fn_orig} -> {fn}")
+    
+    if os.path.exists(fn_orig):
+        if os.path.exists(fn):
+            print(f"both {fn} and {fn_orig} exist, resolve the conflict manually first ..." )
+            raise Exception
+        fdir = os.path.dirname(fn)
+        if not os.path.exists(fdir):
+            makedirs(fdir, mode=0o2775)
```
Comment (mrakitin) [inline startup/39-original_suitcase.py:52]: Is `makedirs` imported somewhere above?

File: startup/39-original_suitcase.py [hunk 2]
```diff
@@ -28,14 +26,45 @@ def conv_to_list(d):
...
+        if debug:
+            print(f"copying {fn_orig} to {fdir}")
+        tfn = fn+"_partial"
+        shutil.copy(fn_orig, tfn)
+        os.rename(tfn, fn)
+        os.remove(fn_orig)
+    elif not os.path.exists(fn):
+        print(f"could not location the resource at either {fn} or {fn_orig} ...")
```
Comment (mrakitin) [inline startup/39-original_suitcase.py:60]:
```suggestion
        print(f"could not locate the resource at either {fn} or {fn_orig} ...")
```

File: startup/39-original_suitcase.py [hunk 3]
```diff
@@ -28,14 +26,45 @@ def conv_to_list(d):
...
+        raise Exception
+    
+    hf5 = h5py.File(fn, "r")
+    return hf5,hf5["/entry/data/data"]
```
Comment (mrakitin) [inline startup/39-original_suitcase.py:64]:
```suggestion
    return hf5, hf5["/entry/data/data"]
```

File: startup/39-original_suitcase.py [hunk 4]
```diff
@@ -119,14 +153,19 @@ def hdf5_export(headers, filename,
                 events = list(header.events(stream_name=descriptor['name'], fill=False))
 
                 res_dict = {}
-                if "filled" in events[0].keys():
-                    for k in list(events[0]['filled'].keys()):
+                for k,v in list(events[0]['data'].items()):
```
Comment (mrakitin) [inline startup/39-original_suitcase.py:156]:
```suggestion
                for k, v in list(events[0]['data'].items()):
```

File: startup/39-original_suitcase.py [hunk 5]
```diff
@@ -119,14 +153,19 @@ def hdf5_export(headers, filename,
...
+                    if not isinstance(v,str):
```
Comment (mrakitin) [inline startup/39-original_suitcase.py:157]:
```suggestion
                    if not isinstance(v, str):
```

File: startup/39-original_suitcase.py [hunk 6]
```diff
@@ -148,22 +187,25 @@ def hdf5_export(headers, filename,
                                                 fletcher32=True)
 
                     if key in list(res_dict.keys()):
-                        res = res_docs[res_dict[key][0]] 
+                        res = res_docs[res_dict[key][0]]
+                        print(f"preocessing resource ...\n", res)
```
Comment (mrakitin) [inline startup/39-original_suitcase.py:191]:
```suggestion
                        print(f"processing resource ...\n", res)
```

File: startup/39-original_suitcase.py [hunk 7]
```diff
@@ -148,22 +187,25 @@ def hdf5_export(headers, filename,
...
+
                         if res['spec'] == "AD_HDF5" and bulk_h5_res:
                             rawdata = None
                             N = len(res_dict[key])
+                            print(f"copying data from source h5 file(s) directly, N={N} ...")
                             if N==1:
-                                res = res_docs[res_dict[key][0]]
-                                hf5 = h5py.File(res["root"]+update_res_path(res["resource_path"], replace_res_path), "r")
-                                data = hf5["/entry/data/data"]
+                                hf5,data = locate_h5_resource(res_docs[res_dict[key][0]], replace_res_path=rp, debug=debug)
```
Comment (mrakitin) [inline startup/39-original_suitcase.py:202]:
```suggestion
                                hf5, data = locate_h5_resource(res_docs[res_dict[key][0]], replace_res_path=rp, debug=debug)
```

File: startup/39-original_suitcase.py [hunk 8]
```diff
@@ -148,22 +187,25 @@ def hdf5_export(headers, filename,
...
                                 data_group.copy(data, key)
                                 hf5.close()
                                 dataset = data_group[key]
                             else: # ideally this should never happen, only 1 hdf5 file/resource per scan
                                 for i in range(N):
-                                    res = res_docs[res_dict[key][i]]
-                                    hf5 = h5py.File(res["root"]+update_res_path(res["resource_path"], replace_res_path), "r")
-                                    data = hf5["/entry/data/data"]
+                                    hf5,data = locate_h5_resource(res_docs[res_dict[key][i]])
```
Comment (mrakitin) [inline startup/39-original_suitcase.py:208]:
```suggestion
                                    hf5, data = locate_h5_resource(res_docs[res_dict[key][i]])
```

File: startup/39-original_suitcase.py [hunk 9]
```diff
@@ -172,9 +214,11 @@ def hdf5_export(headers, filename,
                                     dataset[i,:] = data
                                     hf5.close()
                         else:
+                            print(f"getting resource data using handlers ...")
                             rawdata = header.table(stream_name=descriptor['name'], 
                                                    fields=[key], fill=True)[key]   # this returns the time stamps as well
                     else:
+                        print(f"compiling resource data from individual envents ...")
```
Comment (mrakitin) [inline startup/39-original_suitcase.py:221]:
```suggestion
                        print(f"compiling resource data from individual events ...")
```

File: startup/40-hdf5.py [hunk 10]
```diff
@@ -196,8 +220,7 @@ def send_to_packing_queue(uid, data_type, froot=data_file_path.gpfs, move_first=
             print(f"scan {uid} was not successful.")
             return 
 
-    PilatusCBFHandler.froot = data_file_path[froot.name]
-    threading.Thread(target=pack_and_move, args=(data_type,uid,proc_path,move_first,)).start() 
+    threading.Thread(target=pack_and_process, args=(data_type,uid,proc_path,move_first,)).start()
```
Comment (mrakitin) [inline startup/40-hdf5.py:223]: According to the update below on line 242, the `move_first` is not an argument anymore.
```suggestion
    threading.Thread(target=pack_and_process, args=(data_type, uid, proc_path,)).start() 
```

File: startup/40-hdf5.py [hunk 11]
```diff
@@ -303,24 +283,15 @@ def pack_and_move(data_type, uid, dest_dir, move_first=True):
                 dt.load_data(debug="quiet")
             dt.fh5.close()
             del dt,dt_exp            
-            if fh5_name is not "tmp.h5":  # temporary fix, for some reason other processes cannot open the packed file
+            if fh5_name!="tmp.h5":  # temporary fix, for some reason other processes cannot open the packed file
```
Comment (mrakitin) [inline startup/40-hdf5.py:286]:
```suggestion
            if fh5_name != "tmp.h5":  # temporary fix, for some reason other processes cannot open the packed file
```

File: startup/40-hdf5.py [hunk 12]
```diff
@@ -303,24 +283,15 @@ def pack_and_move(data_type, uid, dest_dir, move_first=True):
...
                 os.system(f"cd {dest_dir} ; cp tmp.h5 {fh5_name} ; rm tmp.h5")
-            try:
-                gen_report(fh5_name)
-            except:
-                pass
+            if data_type=="sol":
```
Comment (mrakitin) [inline startup/40-hdf5.py:288]:
```suggestion
            if data_type == "sol":
```

File: startup/40-hdf5.py [hunk 13]
```diff
@@ -381,10 +343,8 @@ def process_packing_queue():
             if db[uid].stop['exit_status'] != 'success': # the scan actually finished
                 print(f"scan {uid} was not successful.")
                 return 
-        #print(uid)
-        #print(path)
-        PilatusCBFHandler.froot = data_file_path[frn]
-        threading.Thread(target=pack_and_move, args=(data_type,uid,path,move_first,)).start() 
+
+        threading.Thread(target=pack_and_process, args=(data_type,uid,path,move_first,)).start()
```
Comment (mrakitin) [inline startup/40-hdf5.py:347]: Check my comment about it above.
```suggestion
        threading.Thread(target=pack_and_process, args=(data_type, uid, path,)).start() 
```

**Review by mrakitin — COMMENTED** [review]
Looks good overall. Spotted a few styling issues and the function signature inconsistency.

File: startup/02-vars.py [hunk 14]
```diff
@@ -1,2 +1,20 @@
-current_cycle = '2022-1'
-pilatus_data_dir = "/exp_path/hdf"
+from enum import Enum
+
+class data_file_path(Enum):
```
[the remark this answers is not in the document]
Comment (JunAishima) [reply startup/02-vars.py:3]: I see what you mean - it's an enum (not a true class), so makes more sense as snake case

File: startup/20-pilatus.py [hunk 15]
```diff
@@ -272,10 +239,8 @@ def unstage(self, timeout=5):
                 st = self.cam.acquire.set(0)
                 print(f"force stop {self.name}")
         print(" unarmed.")
-        
-        
-        
-        if self.parent.trigger_mode is PilatusTriggerMode.soft:  
+
+       	if self.parent.trigger_mode is PilatusTriggerMode.soft:
```
[the remark this answers is not in the document]
Comment (JunAishima) [reply startup/20-pilatus.py:243]: agree, the columns appear to line up in between 'print " unarmed.") and 'if self.parent..."

File: startup/20-pilatus.py [hunk 16]
```diff
@@ -435,9 +399,9 @@ def trigger(self):
             # ext: set up callback to clear status after the end-of-exposure
             threading.Timer(self.trig_wait, self._status._finished, ()).start()
         # should advance the file number in external trigger mode???
-        
-        return self._status
-    
+
+       	return self._status
```
[the remark this answers is not in the document]
Comment (JunAishima) [reply startup/20-pilatus.py:403]: I see no problems in the actual code, though agree it doesn't look correct in the reviewer. resolving.

File: startup/39-original_suitcase.py (hunk 1 above)
Comment (JunAishima) [reply startup/39-original_suitcase.py:55]: yes, it is in 02-utils - there is a note that os.makedirs doesn't work regarding permissions

File: startup/03-security.py [hunk 17]
```diff
@@ -106,7 +110,15 @@ def print_time_callback(name, doc):
                 print("#STARTDOC : {}".format(time.ctime(t1)))
         RE.subscribe(print_time_callback)
 
-
+def get_IOC_datapath(ioc_name, substitute_path=None):
+    if data_path=="":
```
[the remark this answers is not in the document]
Comment (JunAishima) [reply startup/03-security.py:120]: no, looks like this is a global defined in line 14

File: startup/39-original_suitcase.py (hunk 2 above)
Comment (JunAishima) [reply startup/39-original_suitcase.py]: fixed in https://github.com/NSLS-II-LIX/profile_collection/pull/13/commits/c15dddc5150db915a5703155dbfa6243e4c6a666

File: startup/40-hdf5.py (hunk 10 above)
Comment (JunAishima) [reply startup/40-hdf5.py]: the code is now correct (no move_first argument)

File: startup/40-hdf5.py (hunk 13 above)
Comment (JunAishima) [reply startup/40-hdf5.py]: fixed in commit https://github.com/NSLS-II-LIX/profile_collection/pull/13/commits/9cd622d019226132fe7ada84779fa970e27a3ed2

File: startup/33-CBFhandler.py [hunk 18]
```diff
@@ -1,122 +0,0 @@
...
-            for i in range(self._fpp):
-                fn = self._template % (self._path, self._filename, start, point_number+i) 
-                ret.append(self.get_data(fn))
-        elif self.trigger_mode==triggerMode.external_trigger:
-            fn = self._template % (self._path, self._filename, start, point_number)
-            ret.append(self.get_data(fn))
-        
-        return np.array(ret).squeeze()
-
-db.reg.register_handler('AD_CBF', PilatusCBFHandler, overwrite=True)
```
[the remark this answers is not in the document]
Comment (JunAishima) [reply startup/33-CBFhandler.py:122]: just heard it is not used anymore!

File: startup/40-hdf5.py (hunk 13 above)
Comment (JunAishima) [reply startup/40-hdf5.py]: I believe this was fixed as part of https://github.com/NSLS-II-LIX/profile_collection/pull/13/commits/9cd622d019226132fe7ada84779fa970e27a3ed2

File: startup/03-security.py [hunk 19]
```diff
@@ -1,16 +1,33 @@
...
 run_id = None
 data_path = ""
-collection_lock_file = "/nsls2/xf16id1/.lock"
-okay_to_move_file = "/nsls2/xf16id1/.okay_to_move"
+collection_lock_file = f"{data_file_path.lustre_legacy.value}/.lock"
 login_time = -1
 
-def login(uname = None, pID = None, rID = None, debug=True, test_only=False,
-          root_path='/nsls2/xf16id1', replace_froot=pilatus_data_dir, share_with=[]):
+def check_access(fn):
```
[the remark this answers is not in the document]
Comment (JunAishima) [reply startup/03-security.py:18]: Now available (along with a useful run function) in https://github.com/NSLS-II/nslsii/pull/153

**Review by mrakitin — APPROVED** [review]
Looks good to me.

**Pull request merged**